src/Psl/DateTime/Duration.php

```diff
@@ -674,27 +674,24 @@ public function toString(int $max_decimals = 3): string
         $sec_sign = $this->seconds < 0 || $this->nanoseconds < 0 ? '-' : '';
         $sec = Math\abs($this->seconds);
 
-        /** @var list<array{string, string}> $values */
-        $values = [
-            [((string) $this->hours), 'hour(s)'],
-            [((string) $this->minutes), 'minute(s)'],
-            [$sec_sign . ((string) $sec) . $decimal_part, 'second(s)'],
-        ];
 
-        // $end is the sizeof($values), use static value for better performance.
-        $end = 3;
-        while ($end > 0 && $values[$end - 1][0] === '0') {
```

Comment:
mutating the while's results in timeouts. I decided to go for a couple of simple if statements instead that results in the same behaviour.